src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -279,7 +553,63 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
     result
   }
 
-  /** Map */
+  /** Computes a prefix scan of the elements of the collection.
+    *
+    *  Note: The neutral element `z` may be applied more than once.
+    *
+    *  @tparam B         element type of the resulting collection
+    *  @param z          neutral element for the operator `op`
+    *  @param op         the associative operator for the scan
+    *
+    *  @return           a new $coll containing the prefix scan of the elements in this $coll
+    */
+  def scan[B >: A](z: B)(op: (B, B) => B): CC[B] = scanLeft(z)(op)
+
+  /** Produces a collection containing cumulative results of applying the
+    *  operator going left to right.
+    *
+    *  $willNotTerminateInf
+    *  $orderDependent
+    *
+    *  @tparam B      the type of the elements in the resulting collection
+    *  @param z       the initial value
+    *  @param op      the binary operator applied to the intermediate result and the element
+    *  @return        collection with intermediate results
+    */
+  def scanLeft[B](z: B)(op: (B, A) => B): CC[B] = fromIterable(View.ScanLeft(coll, z, op))
+
+  /** Produces a collection containing cumulative results of applying the operator going right to left.
+    *  The head of the collection is the last cumulative result.
+    *  $willNotTerminateInf
+    *  $orderDependent
+    *
+    *  Example:
+    *  {{{
+    *    List(1, 2, 3, 4).scanRight(0)(_ + _) == List(10, 9, 7, 4, 0)
+    *  }}}
+    *
+    *  @tparam B      the type of the elements in the resulting collection
+    *  @param z       the initial value
+    *  @param op      the binary operator applied to the intermediate result and the element
+    *  @return        collection with intermediate results
+    */
+  def scanRight[B](z: B)(op: (A, B) => B): CC[B] = {
+    var scanned = z :: immutable.Nil
```

Comment:
Which collection type should we use instead? I think `List` has the fastest `cons` implementation (modulo the effects due to garbage collection…).